Create SDD for CdhCore subtopology

LeStarch opened the pull request:
| | |
|:---|:---|
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |
|**_Generative AI was used in this contribution (y/n)_**|  |

---
## Change Description

SDD for CdhCore

## Rationale

Needs an SDD.

## Future Work

Other sub topologies

## AI Usage (see [policy](https://github.com/nasa/fprime/blob/devel/AI_POLICY.md))

GPT-5 did the SDD (with heavy edits from the contributor).
Co-Pilot wrote the commit message (no edits).

LeStarch committed:
Create SDD for CdhCore subtopology

Added a comprehensive Software Design Document (SDD) for the CdhCore subtopology, detailing requirements, design, usage, configuration, and traceability.

LeStarch committed:
sp

zimri-leisher commented on Svc/Subtopologies/CdhCore/docs/sdd.md:
```diff
@@ -0,0 +1,165 @@
+
+All other services are registered in the calling topology.
+
+Perfect — let’s add the **Design (what the subtopology needs to function)** and **Usage (how to integrate it)** sections. I’ll keep them aligned with the structure we’ve been building.
```
forgot to remove this

kevin-f-ortega replied:
Agreed. Delete this

kevin-f-ortega commented on Svc/Subtopologies/CdhCore/docs/sdd.md:
```diff
@@ -0,0 +1,165 @@
+
+* **Rate Groups**
+
+  * The active components instances (`tlmSend`, and `$health`) require scheduling via rate group ports.
```
`tlmSend` is not listed as an active component in the table able. That is because `tlmSend` is configurable. So I would not list `tlmSend` here unless you meant the default `tlmSend`

kevin-f-ortega commented on Svc/Subtopologies/CdhCore/docs/sdd.md:
````diff
@@ -0,0 +1,165 @@
+```fpp
+topology Flight {
+  import CdhCore.Subtopology
+
````
Should add `import ComCcsds.Subtopology` here

kevin-f-ortega commented on Svc/Subtopologies/CdhCore/docs/sdd.md:
```diff
@@ -0,0 +1,165 @@
+
+## 4. Configuration
+
+> This section summarizes the **knobs defined in the CdhCore configuration module**. Users are expected to know how to override defaults in their project config layer.
```
Is there a document that explains how to do this? Perhaps adding a link to that document here would help.

kevin-f-ortega reviewed (changes requested):
LGTM. A few minor changes

kevin-f-ortega commented on Svc/Subtopologies/CdhCore/docs/sdd.md:
```diff
@@ -0,0 +1,165 @@
+| SVC-CDHCORE-006 | The subtopology shall provide **assert to fatal handling functionality** to convert `FW_ASSERT`s into `FATAL` events.         | Inspection |
+| SVC-CDHCORE-007 | The subtopology shall provide **fatal-event routing functionality** to forward fatal announcements to a configurable handler. | Inspection |
+| SVC-CDHCORE-008 | The subtopology shall provide **telemetry sending functionality** through a configurable telemetry interface.                 | Inspection |
+| SVC-CDHCORE-009 | The subtopology shall support **configurable instance properties** for IDs, queue sizes, stack sizes, and priorities.         | Inspection |
```
Add requirement for `The subtopology shall expose rate-group connection points for any rate-drive components it contains.`

LeStarch committed:
Update SDD with review feedback

Added section for rate-group connection points and updated instance requirements.

LeStarch commented:
@kevin-f-ortega done!
